projects/wave-core-new/src/lib/operators/datatype.model.ts

```diff
@@ -254,29 +255,50 @@ export abstract class VectorDataType {
      * @return The name of the data type.
      */
     abstract getCode(): string;
+
+    /**
+     * @return the corresponding result type
+     */
+    abstract toResultType(): ResultType;
```

Comment:
Why don't you use a readonly public variable of type ResultType and call it `resultType`?
I guess there is no need for a function.

Reply:
94609d6c441271ef765979abea31b424f563ba74